llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -4368,6 +4368,8 @@ void OverflowTracking::mergeFlags(Instruction &I) {
     HasNUW &= I.hasNoUnsignedWrap();
     HasNSW &= I.hasNoSignedWrap();
   }
+  if (auto *DisjointOp = dyn_cast<PossiblyDisjointInst>(&I))
```

Comment:
This adds a really subtle usage error - imagine you call mergeFlags on first an add, and then a or.  Do we need to have overflow tracking keep track of the instruction opcode it's being used on?  Or alternative, promote the disjoint to the corresponding meaning on the add (in my previous example)?

(This is okay as a follow up, the current usage doesn't hit this case.)

Reply:
Thanks, I put up https://github.com/llvm/llvm-project/pull/141231 to add an assert, thanks!